xds/internal/xdsclient/authority.go

```diff
@@ -306,6 +333,29 @@ func (a *authority) newConnectionError(err error) {
 	// from the transport layer.
 	for _, rType := range a.resources {
 		for _, state := range rType {
+			// If the connection/stream breaks after the resource has been requested,
+			// but before the response is received, we stop the timer here. When the stream
+			// is recreated, the transport layer will resend all previously requested resources,
+			// and we will start the timer in the `onResourceSend` callback.
+			if state.wState == watchStateRequested {
+				state.wTimer.Stop()
+				state.wState = watchStateStarted
+			}
+		}
+	}
+
+	// Do not propagate errors to watchers if error is of type IgnoredAdsErrorType,
```

Comment:
This comment needs updating since there is no type `IgnoredAdsErrorType` now.

Something like this would be better I feel:
```
// We do not consider it an error if the ADS stream was closed after having received 
// a response on the stream. This is because there are legitimate reasons why the server
// may need to close the stream during normal operations, such as needing to rebalance
// load or the underlying connection hitting its max connection age limit.
// See gRFC A57 for more details.
```